### PR TITLE
`enforceMacOSAppLocation`: showing an error when another version of the app is already running

### LINKED ISSUE #13: enforceMacOSAppLocation doesn't work if the old app is currently open
`enforceMacOSAppLocation` fails in this scenario:

- Have Foo v1 located in `/Applications/Foo.app`
- Open it
- Have Foo v2, still named `Foo.app`, located somewhere else
- Open it
- It will ask the user to move it into `/Applications`
- Instead of doing that, `enforceMacOSAppLocation` will focus to the currently open Foo v1 window

I suppose the actual moving of the `.app` file failed silently but this library was stil trying to open the app and the previously opened one gained the focus this way?

I guess in this situation another dialog should be shown asking the user to close the old app and retrying later.

### PR DESCRIPTION
**Pull request by fabiospampinato**
Closes #13.

The linter isn't working properly as it's throwing an error coming from a file inside `node_modules`, so there are probably some errors in this PR according to your linter.

### CODE REVIEW

**Comment by fabiospampinato** [comment]
The linter seems to be working properly inside travis, weird, I'll address those issues myself then.

File: source/enforce-macos-app-location.js [hunk 1]
```diff
@@ -30,5 +30,18 @@ module.exports = () => {
 		return;
 	}
 
-	api.app.moveToApplicationsFolder();
+	api.app.moveToApplicationsFolder({
+		conflictHandler: conflict => {
+			if (conflict === 'existsAndRunning') { // Can't replace the active version of the app
+				api.dialog.showMessageBoxSync({
+					type: 'error',
+					message: `Another version of ${appName} is currently running, quit it and retry.`,
```
Comment (sindresorhus) [inline source/enforce-macos-app-location.js]: "Retry" what? Not clear enough what the user is supposed to do.

```suggestion
					message: `Another version of ${appName} is currently running. Quit the running app and then launch this version of the app again.`,
```

How about this?

File: source/enforce-macos-app-location.js [hunk 2]
```diff
@@ -30,5 +30,18 @@ module.exports = () => {
...
+					buttons: [`Quit ${appName}`]
```
Comment (sindresorhus) [inline source/enforce-macos-app-location.js]: This makes it ambiguous what you're "quitting". The running app or this app. I think just "OK" would be clear enough in this case.

File: source/enforce-macos-app-location.js (hunk 1 above)
Comment (fabiospampinato) [reply source/enforce-macos-app-location.js]: "quit it" seems fairly clear to me, expanding it to "Quit the running app" sounds too verbose.

"retry" can be definitely more obscure than "then launch this version of the app again", but it's also a lot less text to read 🤔

Maybe a middle ground "Another version of ${app.getName ()} is currently running. Quit it, then launch this version of the app again." would work best.

Let's go with whatever you like best.

File: source/enforce-macos-app-location.js (hunk 2 above)
Comment (fabiospampinato) [reply source/enforce-macos-app-location.js]: I think you are right 🤔 

Another option would adding an "OK" button and a "Quit & Retry" button, or something like that, that kills the other app and reopens the current one automatically. I'm not sure if that's a good idea though.

Comment (fabiospampinato) [reply source/enforce-macos-app-location.js]: > Another option would be adding an "OK" button and a "Quit & Retry" button, or something like that, that kills the other app and reopens the current one automatically. I'm not sure if that's a good idea though.

If one's willing to take that route maybe we may try to kill the other app in the background and retry transparently without even showing a dialog.

File: source/enforce-macos-app-location.js (hunk 1 above)
Comment (sindresorhus) [reply source/enforce-macos-app-location.js]: > Maybe a middle ground "Another version of ${app.getName ()} is currently running. Quit it, then launch this version of the app again." would work best.

👍

File: source/enforce-macos-app-location.js (hunk 2 above)
Comment (sindresorhus) [reply source/enforce-macos-app-location.js]: I don't think quitting an app silently is a good idea. They might have important unsaved work or something.

I think just making it `OK` is the best solution. This is an edge-case anyway.

**Comment by sindresorhus** [comment]
Can you test that this actually works as it should?

**Comment by fabiospampinato** [comment]
I've been using this code in [Notable](https://github.com/notable/notable/releases) since v1.8.2 and it's been working for me:

image

No errors reported about this either.

I've also just checked if it throws an error in Electron v6, where `moveToApplicationsFolder`'s options object wasn't implemented yet, and it doesn't throw an error.

**Comment by sindresorhus** [comment]
Great! :)
